prov/rxm/src/rxm_cq.c

```diff
@@ -240,6 +240,19 @@ static inline int rxm_finish_send(struct rxm_tx_entry *tx_entry)
 	return rxm_finish_send_nobuf(tx_entry);
 }
 
+static inline int rxm_finish_sar_send(struct rxm_tx_entry *tx_entry)
+{
+	struct rxm_tx_buf *tx_buf;
+
+	dlist_pop_front(&tx_entry->in_flight_tx_buf_list, struct rxm_tx_buf,
+			tx_buf, in_flight_entry);
+
+	rxm_tx_buf_release(tx_entry->ep, tx_buf);
+	if (!--tx_entry->segs_left)
```

Comment:
Having 'finish' in the function name implies to me that we're done and simply completing the transfer.  Why is there a check for segs_left here?  Please at least describe what this patch is doing in the commit message.  If segs_left can be > 0, maybe add a comment as to why if it isn't obvious.

Or maybe 'finish' isn't the right word for the function name and corresponding state?

Reply:
yes, I think `rxm_finish_sar_segment_send` is more better name here.
I've updated commit message with information what this patch does